### PR TITLE
Adding horovod

### LINKED ISSUE #6228: Package horovod
Would be great to package [`horovod`]( https://github.com/uber/horovod ), a "distributed training framework for TensorFlow, Keras, and PyTorch".

### PR DESCRIPTION
**Pull request by njzjz**
Resolves #6228.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by njzjz** [comment]
Blocked by https://github.com/horovod/horovod/issues/3956

**Comment by njzjz** [comment]
@conda-forge-admin please restart ci

**Comment by njzjz** [comment]
@conda-forge/help-python-c ready for review.

The package is built against both tensorflow and pytorch, but one is free to install only one of them.

File: recipes/horovod/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+build:
+  number: 0
+  skip: true  # [win]
+  string: {{ proc_type }}_mpi_{{ mpi }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}
+  ignore_run_exports_from:
+    - flatbuffers
+    - tensorflow
+    - pytorch
+  missing_dso_whitelist:
```
Comment (xhochy) [inline recipes/horovod/meta.yaml:21]: Please add a comment here why they are listed.

Comment (njzjz) [reply recipes/horovod/meta.yaml:28]: Added.

File: recipes/horovod/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}  # [cuda_compiler_version != 'None']
+    - cmake >=3.13
+    - make  # [unix]
+    - python  # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+
```
Comment (xhochy) [inline recipes/horovod/meta.yaml:45]: You are using `flatc` directly in the build
```suggestion
   - flatbuffers
```

**Review by xhochy — APPROVED** [review]

**Pull request merged**